src/v/cloud_storage/segment_state.h

```diff
@@ -66,7 +48,7 @@ struct materialized_segment_state {
 
     ss::future<> stop();
 
-    offloaded_segment_state offload(remote_partition* partition);
+    void offload(remote_partition* partition);
```

Comment:
do we still need `base_rp_offset` and `offset_key` in this struct considering that we can take them from `segment`?

Reply:
not really